docs/en/upload.md

```diff
@@ -56,21 +56,63 @@ function startup()
     end
 end
 
+-- Define WiFi station event callbacks 
+wifi_connect_event = function(T) 
+  print("Connection to AP("..T.SSID..") established!")
+  print("Waiting for ip...") 
+end
+
+wifi_got_ip_event = function(T) 
+  print("Wifi connection is ready! IP address is: "..T.IP)
+  print("Startup will resume momentarily, you have 3 seconds to abort.")
+  print("Waiting...") 
+  tmr.create():alarm(3000, tmr.ALARM_SINGLE, startup)
+end
+
+wifi_disconnect_event = function(T)
+  if T.reason == wifi.eventmon.reason.ASSOC_LEAVE then 
+    --the station has disassociated from a previously connected AP
+    return 
+  end
+  -- total_tries: how many times the station will attempt to connect to the AP.
+  local total_tries = 3
+  local reason_string=""
+  print("\nWiFi connection to AP("..T.SSID..") has failed!")
+
+  --There are many possible disconnect reasons, the following iterates through 
+  --the list and returns the string corresponding to the disconnect reason.
+  for key,val in pairs(wifi.eventmon.reason) do
+    if val == T.reason then
+      print("Disconnect reason:"..T.reason.."("..key..")")
+      reason_string=key
```

Comment:
I was going to add the disconnect reason to the final abort message, but I decided not to and I forgot to remove this variable.